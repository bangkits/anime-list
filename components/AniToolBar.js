/** @jsxImportSource @emotion/react */

import React from 'react';
import AppBar from '@mui/material/AppBar';
import useScrollTrigger from '@mui/material/useScrollTrigger';

import ArrowBack from "@mui/icons-material/ArrowBack";
import { useRouter } from 'next/router';
import { Typography } from '@mui/material';

function ElevationScroll(props) {
  const { children, window } = props;
  // Note that you normally won't need to set the window ref as useScrollTrigger
  // will default to window.
  // This is only being set here because the demo is in an iframe.
  const trigger = useScrollTrigger({
    disableHysteresis: true,
    threshold: 0,
    target: window ? window() : undefined,
  });

  return React.cloneElement(children, {
    elevation: trigger ? 4 : 0,
  });
}

const ToolBar = (props) => {
  const { hasBackButton, title } = props;
  const { back } = useRouter();

  return (
    <ElevationScroll {...props}>
      <AppBar css={{
        background: 'linear-gradient(to right top, #00F556, #60B6F1)',
        color: 'black',
        display: 'flex',
        flexDirection: 'row',
        alignItems: 'center',
        minHeight: 75,
        padding: 10,
        flex: 1,
        borderBottomRightRadius: 5,
        borderBottomLeftRadius: 5
      }}>
        {hasBackButton && <ArrowBack aria-label="arrow-back" css={{ color: 'white' }} onClick={() => back()} />}
        {title && <Typography css={{ marginLeft: 20, color: 'white' }} variant="h4">{title}</Typography>}
        {props.children}
      </AppBar>
    </ElevationScroll>
  )
};

export default ToolBar;