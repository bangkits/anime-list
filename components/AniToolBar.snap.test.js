import renderer from 'react-test-renderer';
import AniToolBar from './AniToolBar';

jest.mock('next/router', () => ({
  useRouter() {
    return ({
      back: jest.fn()
    });
  },
}));

describe('AniToolbar Component Snapshot', () => {
  it('Renders AniToolbar with back button and title', () => {
    const props = {
      hasBackButton: true,
      title: 'Title'
    };

    const tree = renderer
      .create(<AniToolBar {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders AniToolbar with title but without back button', () => {
    const props = {
      hasBackButton: false,
      title: 'Title'
    };

    const tree = renderer
      .create(<AniToolBar {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders AniToolbar with back button but without title', () => {
    const props = {
      hasBackButton: true
    };

    const tree = renderer
      .create(<AniToolBar {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders AniToolbar without back button and title', () => {
    const props = {};

    const tree = renderer
      .create(<AniToolBar {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});