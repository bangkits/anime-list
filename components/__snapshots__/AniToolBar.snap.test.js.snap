// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AniToolbar Component Snapshot Renders AniToolbar with back button and title 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1f8m4u1-MuiPaper-root-MuiAppBar-root-ToolBar"
>
  <svg
    aria-hidden={true}
    aria-label="arrow-back"
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ofg42n-MuiSvgIcon-root-ToolBar"
    data-testid="ArrowBackIcon"
    focusable="false"
    onClick={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
    />
  </svg>
  <h4
    className="MuiTypography-root MuiTypography-h4 css-1pi1kz0-MuiTypography-root-ToolBar"
  >
    Title
  </h4>
</header>
`;

exports[`AniToolbar Component Snapshot Renders AniToolbar with back button but without title 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1f8m4u1-MuiPaper-root-MuiAppBar-root-ToolBar"
>
  <svg
    aria-hidden={true}
    aria-label="arrow-back"
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ofg42n-MuiSvgIcon-root-ToolBar"
    data-testid="ArrowBackIcon"
    focusable="false"
    onClick={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
    />
  </svg>
</header>
`;

exports[`AniToolbar Component Snapshot Renders AniToolbar with title but without back button 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1f8m4u1-MuiPaper-root-MuiAppBar-root-ToolBar"
>
  <h4
    className="MuiTypography-root MuiTypography-h4 css-1pi1kz0-MuiTypography-root-ToolBar"
  >
    Title
  </h4>
</header>
`;

exports[`AniToolbar Component Snapshot Renders AniToolbar without back button and title 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1f8m4u1-MuiPaper-root-MuiAppBar-root-ToolBar"
/>
`;
